internal/argotunnel/informer.go

```diff
@@ -103,8 +103,10 @@ func ingressSecretIndexFunc(ingressClass string, secret *resource) func(obj inte
 					if rule.HTTP != nil && len(rule.Host) > 0 {
 						if r, ok := hostsecret[rule.Host]; ok {
 							idx = append(idx, itemKeyFunc(r.namespace, r.name))
-						} else if secret != nil {
-							idx = append(idx, itemKeyFunc(secret.namespace, secret.name))
+						} else {
+							for _, secret := range secrets {
+								idx = append(idx, itemKeyFunc(secret.namespace, secret.name))
```

Comment:
this is actually very different, the the previous else.
- old version - no secret was specified, the default was provided, and therefore the ingress will use that secret and it makes sense to index the secret against the ingress (creating a graph relationship)
- new version - no secret was specified, all the defaults will not be used, however all will be indexed (creating graph relationships where non-exist).

Why is this bad?  When graph relationships change the ingress controller needs to reconfigure itself.  Here, if any of the default secrets are deleted (even those that are not used). In this case, all ingresses tied to any default secret will be deleted with the secret, even though the ingress does not use the secret.